### PR TITLE
Put correlation id in log4j2 thread context (more on thread context h…

### PR DESCRIPTION
**Pull request by geiralund**
…ere https://logging.apache.org/log4j/2.x/manual/thread-context.html)

### CODE REVIEW

File: src/main/kotlin/no/nav/dagpenger/streams/Pond.kt [hunk 1]
```diff
@@ -22,6 +24,7 @@ abstract class Pond(private val topic: Topic<String, Packet>) : Service() {
                 val timer = processTimeLatency.startTimer()
                 onPacket(packet)
                 timer.observeDuration()
+                ThreadContext.clearAll()
             }
```
Comment (gtcno) [inline src/main/kotlin/no/nav/dagpenger/streams/Pond.kt]: Is this safe? Shouldnt we just remove the correlation key?

File: src/main/kotlin/no/nav/dagpenger/streams/River.kt [hunk 2]
```diff
@@ -38,6 +41,7 @@ abstract class River(private val topic: Topic<String, Packet>) : Service() {
                 }
             }
             .peek { key, packet -> LOGGER.info("Producing packet with key $key and value: $packet") }
+            .peek { _, _ -> ThreadContext.clearAll() }
             .toTopic(topic)
```
Comment (gtcno) [inline src/main/kotlin/no/nav/dagpenger/streams/River.kt]: Ditto

File: src/main/kotlin/no/nav/dagpenger/streams/Pond.kt [hunk 3]
```diff
@@ -24,7 +24,8 @@ abstract class Pond(private val topic: Topic<String, Packet>) : Service() {
                 val timer = processTimeLatency.startTimer()
                 onPacket(packet)
                 timer.observeDuration()
-                ThreadContext.clearAll()
+                ThreadContext.remove(CorrelationId.X_CORRELATION_ID)
+                ThreadContext.remove("vvvv")
```
Comment (gtcno) [inline src/main/kotlin/no/nav/dagpenger/streams/Pond.kt]: vvvv?

**Comment by geiralund** [comment]
Doh...
Skulle bare sjekke å fjerne en ukjent, skulle ikke vært sjekket inn..

16. des. 2019 kl. 20:54 skrev Giao The Cung <notifications@github.com>:

﻿

@gtcno commented on this pull request.

________________________________

In src/main/kotlin/no/nav/dagpenger/streams/Pond.kt<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnavikt%2Fdagpenger-streams%2Fpull%2F37%23discussion_r358433178&data=02%7C01%7Cgeir.andre.lund%40nav.no%7C7073f3d17f7c44cbd0f308d78261bb24%7C623665341ec3496288699b5535279d0b%7C1%7C0%7C637121228573861999&sdata=dWPuHkd60c9LdlQ85ZaNYXE1A31Sx02xzNgLWcTI3R8%3D&reserved=0>:

> @@ -24,7 +24,8 @@ abstract class Pond(private val topic: Topic<String, Packet>) : Service() {
                 val timer = processTimeLatency.startTimer()
                 onPacket(packet)
                 timer.observeDuration()
-                ThreadContext.clearAll()
+                ThreadContext.remove(CorrelationId.X_CORRELATION_ID)
+                ThreadContext.remove("vvvv")


vvvv?

**Pull request merged**